server/src/test/java/org/opensearch/common/blobstore/stream/read/listener/ListenerTestUtils.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.blobstore.stream.read.listener;
+
+import org.opensearch.common.Randomness;
+import org.opensearch.core.action.ActionListener;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Utility class containing common functionality for read listener based tests
+ */
+public class ListenerTestUtils {
+
+    /**
+     * TestInputStream generates a variable length, random input stream for test use cases.
+     */
+    static class TestInputStream extends InputStream {
```

Comment:
Can you replace this with the following?

```
new ByteArrayInputStream(randomByteArrayOfLength(length))
```

Reply:
Updated.